src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -878,6 +881,31 @@ err_t sockinfo_tcp::ip_output_syn_ack(struct pbuf *p, void* v_p_conn, int is_rex
 {
 	sockinfo_tcp *p_si_tcp = (sockinfo_tcp *)pcb_container;
 	p_si_tcp->m_p_socket_stats->tcp_state = new_state;
+
```

Comment:
... having the Agent do its handling internally...
by using a callback mechanism
The Agent should register for updates, then when a state changes sockinfo_tcp will callback the Agent